lib/channels/src/index.ts

```diff
@@ -0,0 +1,152 @@
+export type ChannelHandler<TEventArgs = any> = (event: ChannelEvent<TEventArgs>) => void;
```

Comment:
> Also, the whole package is type safe - no `any` usage except for generics as default value (that makes them optional)

Some questions because my knowledge is limited on TS: Is this what you mean with these default values for generics?  Why exactly do we need to provide a default value for a generic type for the args of this `ChannelHandler`? Sorry if this is answered somewhere else already (since I know that the TS channel is active), but just asking 😄

Reply:
I thought a lot about the generics and came to the conclusion that we should not use generics for channels because it just won't work. It's error by design

My initial thoughts were that you have 1 event with multiple listeners and all of them share the same data-interface but here's the thing, no one is stopping you from doing this:

```typesciprt
channel.addListener<{ id: string }>('event1', data => data.id);
channel.addListener<{ name: string }>('event1', data => data.name);

// Just won't do, there's no correct interface anymore 
channel.listeners<???>('event1');
```

I will remove them for now, thanks for asking this!